src/dreyfus_index.erl

```diff
@@ -157,9 +159,43 @@ handle_call(info, _From, State) -> % obsolete
     Reply = info_int(State#state.index_pid),
     {reply, Reply, State}.
 
+handle_cast({ddoc_updated, DDocResult}, State) ->
+    #state{dbname = DbName, index=#index{sig=CurrentSig}, waiting_list = WaitList} = State,
+    IndexDeleted = case DDocResult of
+        {not_found, deleted} ->
+            true;
+        {ok, DDoc} ->
+            Index = get_index_with_sig(DDoc, CurrentSig),
+            case Index of
+                [] ->
+                    true;
+                _ ->
```

Comment:
match something explicitly here, a non-empty list, for example. we don't expect any other result from get_index_with_sig, so don't have clauses for them.